samigo/samigo-impl/src/java/org/sakaiproject/samigo/impl/SamigoAvailableNotificationServiceImpl.java

```diff
@@ -0,0 +1,369 @@
+    @Setter private ScheduledInvocationManager scheduledInvocationManager;
+    @Setter private ServerConfigurationService serverConfigurationService;
+    @Setter private SessionManager sessionManager;
+    @Setter private SiteService siteService;
+    @Setter private UserDirectoryService userDirectoryService;
+    @Setter private UserTimeService userTimeService;
+    private PublishedAssessmentService publishedAssessmentService;
+    private static final ResourceLoader rl = new ResourceLoader("SamigoAvailableNotificationMessages");
+    private final List<String> additionalHeaders = new ArrayList<>();
+    //feedback
+    public static Integer NO_FEEDBACK = Integer.valueOf(3);
+    //scoring types
+    public static final Integer HIGHEST_SCORE = 1;
+    public static final Integer LAST_SCORE= 2;
+    public static final Integer ALL_SCORE= 3;
+    public static final Integer AVERAGE_SCORE= 4;
+    //this matches DEAD_STATUS from AssessmentBaseIfc, indicating a deleted assessment
+    public static final Integer DEAD_STATUS = 2;
+
+    public void init() {
+        log.debug("SamigoAvailableNotificationService init()");
+
+        publishedAssessmentService = new PublishedAssessmentService();
+
+        String sender = "Sender: \"" + getServiceName() + "\" <" + getSetupRequest() + ">";
+        additionalHeaders.add(sender);
+        additionalHeaders.add("Content-type: text/html; charset=UTF-8");
+    }
+
+    public void scheduleAssessmentAvailableNotification(String publishedId) {
+
+        // Remove any previously scheduled notification
+        removeScheduledAssessmentNotification(publishedId);
+
+        try {
+            PublishedAssessmentFacade publishedAssessment = publishedAssessmentService.getPublishedAssessment(publishedId);
+            List<ExtendedTime> extensionContainer = PersistenceService.getInstance().getExtendedTimeFacade().getEntriesForPub(publishedAssessment.getData());
+            Date startDate = publishedAssessment.getStartDate();
+            // Only schedule a new notification if start date is in the future to prevent duplicates and spam
+            if (Instant.now().isBefore(startDate.toInstant())) {    //for main
+                scheduledInvocationManager.createDelayedInvocation(startDate.toInstant(), "org.sakaiproject.samigo.api.SamigoAvailableNotificationService", publishedId);
+            }
+            for (ExtendedTime extension: extensionContainer){  //make separate delayedInvocations for people with exceptions.
+                if (Instant.now().isBefore(extension.getStartDate().toInstant())) {
+                    scheduledInvocationManager.createDelayedInvocation(extension.getStartDate().toInstant(), "org.sakaiproject.samigo.api.SamigoAvailableNotificationService", publishedId + ',' + extension.getId().toString());
+                }
+            }
+        } catch (IllegalArgumentException e) {
+            log.error("Unable to schedule assessment open notification for assessment " + publishedId);
```

Comment:
I think you missed this one:
```suggestion
            log.error("Unable to schedule assessment open notification for assessment {}", publishedId);
```